qiskit/utils/deprecate_pulse.py

```diff
@@ -32,25 +32,52 @@ def deprecate_pulse_func(func):
     )(func)
 
 
-def deprecate_pulse_dependency(*args, **kwargs):
-    """Deprecation message for functions and classes which use or depend on Pulse"""
-    decorator = deprecate_func(
-        since="1.3",
-        package_name="Qiskit",
-        removal_timeline="in Qiskit 2.0",
-        additional_msg="The entire Qiskit Pulse package is being deprecated "
-        "and this is a dependency on the package.",
-        **kwargs,
-    )
+def deprecate_pulse_dependency(*args, moving_to_dynamics: bool = False, **kwargs):
+    """Deprecation message for functions and classes which use or depend on Pulse
+
+    moving_to_dynamics: set to True if the dependency is moving to Qiskit Dynamics. This affects
+    the deprecation message being printed, namely saying explicitly whether the dependency will
+    be moved to Qiskit Dynamics or whether it will just be removed without an alternative.
```

Comment:
This will be my final nitpick, I know the docstring is not going to stay, but in case it's used as a reference moving forward for future deprecation decorators:

```suggestion
        Args:
            moving_to_dynamics: set to True if the dependency is moving to Qiskit Dynamics.
                This affects the deprecation message being printed, namely saying explicitly whether 
                the dependency will be moved to Qiskit Dynamics or whether it will just be removed 
                without an alternative.
```

Reply:
Done, thanks.